### PR TITLE
Allow for MemberExpression in JSXIdentifier

### PR DESCRIPTION
**Pull request by RSO**
Tests should cover why I would like to have this.

### CODE REVIEW

File: transforms/create-element-to-jsx.js [hunk 1]
```diff
@@ -60,11 +60,20 @@ module.exports = function(file, api, options) {
     }
   };
 
+  const jsxIdentifierNameFor = node => {
```
Comment (cpojer) [inline transforms/create-element-to-jsx.js]: Can we change this to just return the node itself instead of generating a string?

Comment (RSO) [inline transforms/create-element-to-jsx.js]: Doing that would make the recursion harder to accomplish. The recursion is needed to allow for nested MemberExpressions.

Comment (cpojer) [inline transforms/create-element-to-jsx.js]: why do you need the recursion? Can't you just return "node" if its a MemberExpression and make the code accept either an AST node or a string?

File: transforms/create-element-to-jsx.js [hunk 2]
```diff
@@ -66,11 +66,23 @@ module.exports = function(file, api, options) {
     }
   };
 
+  const jsxIdentifierFor = node => {
```
Comment (RSO) [inline transforms/create-element-to-jsx.js]: @cpojer as you can see, the method now returns a node, but the recursion is still there, as we need to be able to convert `React.createElement(A.b.c.d)` as well. So `node.object` can be another `MemberExpression` that needs to be converted to a `JsxMemberExpression`, and `node.object` is an Identifier that should be converted to a `jsxIdentifier`. I could probably just change that for `j.jsxIdentifier(node.name)` but that would technically be duplicating the `else` block.

I hope I understood your request ;-) let me know what you think!

Comment (cpojer) [inline transforms/create-element-to-jsx.js]: sounds good! I honestly didn't know that jsxMemberExpression was something different from a MemberExpression, that's silly.

**Pull request merged**